geomesa-accumulo/geomesa-accumulo-tools/src/main/scala/org/locationtech/geomesa/accumulo/tools/AccumuloDataStoreCommand.scala

```diff
@@ -18,14 +18,25 @@ trait AccumuloDataStoreCommand extends DataStoreCommand[AccumuloDataStore] {
 
   override def params: AccumuloDataStoreParams
 
-  override def connection: Map[String, String] = Map[String, String](
-    AccumuloDataStoreParams.instanceIdParam.getName -> params.instance,
-    AccumuloDataStoreParams.zookeepersParam.getName -> params.zookeepers,
-    AccumuloDataStoreParams.userParam.getName       -> params.user,
-    AccumuloDataStoreParams.passwordParam.getName   -> params.password,
-    AccumuloDataStoreParams.tableNameParam.getName  -> params.catalog,
-    AccumuloDataStoreParams.visibilityParam.getName -> params.visibilities,
-    AccumuloDataStoreParams.authsParam.getName      -> params.auths,
-    AccumuloDataStoreParams.mockParam.getName       -> params.mock.toString
-  ).filter(_._2 != null)
+  override def connection: Map[String, String] = {
+    val parsedParams = Map[String, String](
+      AccumuloDataStoreParams.instanceIdParam.getName -> params.instance,
+      AccumuloDataStoreParams.zookeepersParam.getName -> params.zookeepers,
+      AccumuloDataStoreParams.userParam.getName       -> params.user,
+      AccumuloDataStoreParams.passwordParam.getName   -> params.password,
+      AccumuloDataStoreParams.tableNameParam.getName  -> params.catalog,
+      AccumuloDataStoreParams.visibilityParam.getName -> params.visibilities,
+      AccumuloDataStoreParams.authsParam.getName      -> params.auths,
+      AccumuloDataStoreParams.mockParam.getName       -> params.mock.toString
+    ).filter(_._2 != null)
+
+    // AccumuloDataStoreFactory requires instance ID to be set but it should not be required if mock is set...so set a
+    // fake one but be careful NOT to add a mock zoo since other code will then think it has a zookeeper but doesn't
+    if (parsedParams.get(AccumuloDataStoreParams.mockParam.getName).exists(_.toBoolean)) {
```

Comment:
Though it may be good to remove the user and password modification in case the user expected something regarding that? I seem to remember though that the standard user/pass combination doesn't work correctly with the mock